File: scripts/screenshot-commission.py
#!/usr/bin/env python3
"""Screenshot the Commission list page using Playwright (Python).

Run from repo root:  python3 scripts/screenshot-commission.py
"""

import os
from pathlib import Path
from playwright.sync_api import sync_playwright

OUT_DIR = Path("/home/ttdiy/aierp/docs/screenshots")
OUT_DIR.mkdir(parents=True, exist_ok=True)
FRONTEND = "http://localhost:3002"
USERNAME = os.getenv("AIERP_LOGIN_USERNAME", "admin")
PASSWORD = os.getenv("AIERP_LOGIN_PASSWORD", "admin123")

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    ctx = browser.new_context(
        viewport={"width": 1440, "height": 900},
        device_scale_factor=2,
        locale="zh-CN",
    )
    page = ctx.new_page()

    print("→ login page")
    page.goto(f"{FRONTEND}/login", wait_until="networkidle")
    page.screenshot(path=str(OUT_DIR / "01-login.png"))
    print("  saved 01-login.png")

    print("→ login as admin")
    page.fill('input[type="text"]', USERNAME)
    page.fill('input[type="password"]', PASSWORD)
    page.click('button[type="submit"]')
    page.wait_for_url(f"{FRONTEND}/", timeout=10000)
    page.wait_for_load_state("networkidle")
    page.screenshot(path=str(OUT_DIR / "02-dashboard.png"))
    print("  saved 02-dashboard.png")

    print("→ /finance/commissions")
    # Pre-warm the page (first load is slow due to lazy chunk + dev compile)
    page.goto(f"{FRONTEND}/finance/commissions", wait_until="domcontentloaded")
    page.wait_for_timeout(3000)
    # Reload to get a clean state with all data fetched
    page.reload(wait_until="networkidle")
    page.wait_for_timeout(2000)
    page.screenshot(path=str(OUT_DIR / "03-commission-list.png"))
    print("  saved 03-commission-list.png")

    print("→ open create drawer")
    try:
        page.click('button:has-text("新建佣金")', timeout=3000)
        page.wait_for_timeout(600)
        page.screenshot(path=str(OUT_DIR / "04-commission-drawer.png"))
        print("  saved 04-commission-drawer.png")
    except Exception as e:
        print(f"  (drawer button not found, skipped: {e})")

    browser.close()

print(f"\n✓ done. screenshots in {OUT_DIR}")

File: scripts/screenshot-commission.mjs
// Screenshot the Commission list page (and login if needed).
// Run from the repo root:  /home/ttdiy/.local/bin/playwright screenshot scripts/screenshot-commission.mjs http://localhost:3002/login docs/screenshots/01-login.png
//
// Or use the runner: node scripts/screenshot-commission.mjs

import { chromium } from "/home/ttdiy/.local/lib/node_modules/playwright/index.mjs";
import { mkdirSync } from "node:fs";

const OUT_DIR = "/home/ttdiy/aierp/docs/screenshots";
mkdirSync(OUT_DIR, { recursive: true });

const FRONTEND = "http://localhost:3002";
const USERNAME = process.env.AIERP_LOGIN_USERNAME ?? "admin";
const PASSWORD = process.env.AIERP_LOGIN_PASSWORD ?? "admin123";

const browser = await chromium.launch({ headless: true });
const ctx = await browser.newContext({
  viewport: { width: 1440, height: 900 },
  deviceScaleFactor: 2,
  locale: "zh-CN",
});
const page = await ctx.newPage();

console.log("→ login page");
await page.goto(`${FRONTEND}/login`, { waitUntil: "networkidle" });
await page.screenshot({ path: `${OUT_DIR}/01-login.png`, fullPage: false });
console.log("  saved 01-login.png");

// Login as admin
console.log("→ login as admin");
await page.fill('input[placeholder*="用户"], input[type="text"]', USERNAME);
await page.fill('input[type="password"]', PASSWORD);
await page.click('button[type="submit"]');
await page.waitForURL(`${FRONTEND}/`, { timeout: 10000 });
await page.waitForLoadState("networkidle");
await page.screenshot({ path: `${OUT_DIR}/02-dashboard.png`, fullPage: false });
console.log("  saved 02-dashboard.png");

// Navigate to commission list
console.log("→ /finance/commissions");
await page.goto(`${FRONTEND}/finance/commissions`, { waitUntil: "networkidle" });
await page.waitForTimeout(800); // let table render
await page.screenshot({ path: `${OUT_DIR}/03-commission-list.png`, fullPage: false });
console.log("  saved 03-commission-list.png");

// Open the create drawer
console.log("→ open create drawer");
try {
  await page.click('button:has-text("新建佣金")', { timeout: 3000 });
  await page.waitForTimeout(500);
  await page.screenshot({ path: `${OUT_DIR}/04-commission-drawer.png`, fullPage: false });
  console.log("  saved 04-commission-drawer.png");
} catch (e) {
  console.log("  (drawer button not found, skipped)");
}

await browser.close();
console.log("\n✓ done. screenshots in", OUT_DIR);
